### PR TITLE
test: add feed edge-case coverage for Codecov follow-ups

### LINKED ISSUE #352: ## [Codecov](https://app.codecov.io/gh/Hiroki-org/OpenShelf/pull/351?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Hiroki-org) Report
## [Codecov](https://app.codecov.io/gh/Hiroki-org/OpenShelf/pull/351?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Hiroki-org) Report
:x: Patch coverage is `92.64706%` with `15 lines` in your changes missing coverage. Please review.
:white_check_mark: All tests successful. No failed tests found.
| [Files with missing lines](https://app.codecov.io/gh/Hiroki-org/OpenShelf/pull/351?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Hiroki-org) | Patch % | Lines |
|---|---|---|
| [apps/web/src/components/feed-button.tsx](https://app.codecov.io/gh/Hiroki-org/OpenShelf/pull/351?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Hiroki-org#diff-YXBwcy93ZWIvc3JjL2NvbXBvbmVudHMvZmVlZC1idXR0b24udHN4) | 87.30% | [8 Missing :warning: ](https://app.codecov.io/gh/Hiroki-org/OpenShelf/pull/351?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Hiroki-org) |
| [apps/api/src/routes/feed.ts](https://app.codecov.io/gh/Hiroki-org/OpenShelf/pull/351?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Hiroki-org#diff-YXBwcy9hcGkvc3JjL3JvdXRlcy9mZWVkLnRz) | 94.26% | [6 Missing and 1 partial :warning: ](https://app.codecov.io/gh/Hiroki-org/OpenShelf/pull/351?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Hiroki-org) |

:loudspeaker: Thoughts on this report? [Let us know!](https://github.com/codecov/feedback/issues/255)

_Originally posted by @codecov[bot] in https://github.com/Hiroki-org/OpenShelf/issues/351#issuecomment-4200786014_

### PR DESCRIPTION
**Pull request by is0692vs**
## Summary
- add user feed 404 test when user is missing
- verify user feed title/author fallback when `displayName` is null
- verify entry author fallback to `OpenShelf` when feed author name is blank
- add org collection feed 404 tests for missing and private collections

## Linked issues
- Closes #373
- Closes #352

<details><summary><h3>Greptile Summary</h3></summary>

フィードルートのエッジケースカバレッジを強化するテストを5件追加しています。ユーザーフィードの 404・`displayName` null フォールバック・空白著者フォールバック・org コレクションフィードの存在しない/非公開コレクション 404 をそれぞれ検証しており、モックの設定・ルートの期待動作との整合性はいずれも正確です。
</details>

<h3>Confidence Score: 5/5</h3>

テストのみの変更であり、本番コードへの影響はなく、マージは安全です。

5つの新規テストはいずれもルートの実際の動作と整合しており、モック設定も正確です。唯一の指摘（フィードレベルの blank author アサーション欠落）は P2 のスタイル提案であり、マージをブロックするものではありません。

特に注意が必要なファイルはありません。

<details open><summary><h3>Vulnerabilities</h3></summary>

セキュリティ上の懸念は見当たりません。変更はテストファイルのみで、本番コードへの影響はありません。
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/api/src/routes/__tests__/feed.test.ts | ユーザーフィード 404・displayName null フォールバック・空白著者フォールバック・org コレクション 404（存在しない/非公開）の5つのテストを追加。モックの呼び出し順序・ルートの動作との整合性はいずれも正しい。フィードレベルの author 要素が空白になるケースがアサートされていない点のみ P2 指摘あり。 |

</details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/api/src/routes/__tests__/feed.test.ts
Line: 303-339

Comment:
**フィードレベルの `<author>` が空白のまま検証されていない**

このテストはエントリレベルの `<author><name>OpenShelf</name></author>` フォールバックを検証していますが、フィードレベルの `<author>` 要素（`buildAtomFeed` が `meta.authorName` を直接使用して生成する部分）は検証されていません。

`buildUserFeedResponse` では `authorName: user.displayName ?? user.name` = `"   "` がそのまま渡され、`buildAtomFeed` は `escapeXml(meta.authorName)` を使うため、フィードレベルには `<author><name>   </name></author>`（空白3文字）が出力されます。一方、`buildPaperEntries` では `fallbackAuthorName.trim() || "OpenShelf"` でフォールバックが機能しています。

この非対称性を明示するアサーションを追加することで、フィードレベルの挙動が未検証であることが明確になります：

```suggestion
        expect(res.status).toBe(200);
        const text = await res.text();
        expect(text).toContain("User Paper");
        expect(text).toContain("<author><name>OpenShelf</name></author>");
        // フィードレベルの <author> も空白名にならないことを確認する場合は以下を追加:
        // expect(text).not.toMatch(/<author><name>\s+<\/name><\/author>/);
```

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test: expand feed route edge-case covera..."](https://github.com/hiroki-org/openshelf/commit/cd68821d9af37c7a3af12a76e07eeb5d9747bcc4) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=27868690)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: apps/api/src/routes/__tests__/feed.test.ts [hunk 1]
```diff
@@ -250,6 +250,94 @@ describe("feed routes", () => {
...
+        const app = await createTestApp();
+        const env = createTestEnv();
+
+        const res = await app.request("http://localhost/feed/users/user-1/atom.xml", {}, env as any);
+
+        expect(res.status).toBe(200);
+        const text = await res.text();
+        expect(text).toContain("User Paper");
+        expect(text).toContain("<author><name>OpenShelf</name></author>");
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/feed.test.ts:340]: 対応しました: フィード全体のauthor名も`trim() || "OpenShelf"`で正規化し、テストに空白authorを禁止するアサーションを追加しました。影響範囲: `feed.ts`の`buildFeedResponse`と`feed.test.ts`の該当ケース。検証: `npm run test --workspace api -- src/routes/__tests__/feed.test.ts`.
